### ISSUE TITLE
fmtQueryCPUModePct is costly, consider adding label matchers

### ISSUE THREAD

**Issue by GiedriusS** [opencost/opencost#1883]
**Describe the bug**

This query is being consistently executed: https://github.com/opencost/opencost/blob/55d77f1de56d47ecadd6f3bd38c76e8fb9a4136a/pkg/costmodel/cluster.go#L896. Because it selects `node_cpu_seconds_total` from _all_ servers, it means that it leads to huge load on the metrics stack. Consider adding selectors such as `{cluster="..."}` or something else to make the load smaller.

**To Reproduce**

1. Run OpenCost

**Expected behavior**

OpenCost not pulling in a bunch of data from unrelated servers

**Screenshots**

N/A

**Which version of OpenCost are you using?**

N/A

**Additional context**

N/A

**Comment by mattray** [comment]
This sounds worthwhile, @AjayTripathy could we get this looked at for the next release?

**Comment by nikovacevic** [comment]
Hi @GiedriusS -- we agree with your concern here, and we've decided to just disable the querying of those metrics by default. See https://github.com/opencost/opencost/pull/2049

Would that satisfy your request here?

**Comment by KipsasJaujoj** [comment]
No, it does not satisfy the request. By default there is no issue but if you want these metrics than you are still getting unfiltered version of it. The filter should be used to reduce prometheus load.

**Comment by nikovacevic** [comment]
Okay, we will have to reassess at a later date.

**Comment by BigValen** [comment]
Reopening, as this is still a serious problem. 

I thought we were a reasonably small user of opencost around 1000 nodes, many with 192 cores. But, it's enough that it's over the 4bn samples (when opencost tries to grab node_cpu_seconds_total for 24h) and when I scope it down to 15mins for testing, it'll still take over 30 seconds (and timeout). Main problem is we don't need per-core stats.

A recording rule, to avg per core, could shrink this by 200x. But we also don't need this collected every 15s, so it makes complete sense to have a downsampling recording rule. But this means we need to be able to point fmtQueryCPUModePct at a different timeseries. 

Would it be OK to send in a PR that allows some configuration of timeseries names ?

**Comment by ameijer** [comment]
I see no issues with this as long as the configuration defaults to the existing values 👍
